### PR TITLE
Update names of exported functions to optimize for teachability

### LINKED ISSUE #91: [FEATURE] Update names of exported functions to optimize for teachability
# Feature Request Description

Naming things is hard! A good guideline for naming functions is that the name of the function gels well with the context in which you would first explain or teach the function's purpose. For example, when you explain that importing data is the first step of data analysis, and the question is, how do you import data from REDCap, a great answer could be: with `import_redcap()`!

When we then explain that the supertibble has multiple data tibbles embedded into it, and the question is how do you extract those tibbles, the answer could be: with `extract_tibbles()`! Or `extract_tibble()` if you just want one data tibble. Or `bind_tibbles()` if you want to bind the data tibbles to your environment.

We will want to make sure the old functions still work but also let users know that they are now deprecated and will go away soon. Here are two resources for deprecating things smoothly.

- https://r-pkgs.org/lifecycle.html
- https://lifecycle.r-lib.org/articles/communicate.html

This is also a good opportunity to clean up the API a bit. See details below.

Note that roxygen documentation, README, vignettes, as well as screenshots/GIFs will need updating to reflect these changes.

We will also want to add a paragraph to the blog post for 0.2 to explain why we're changing the API.

# Proposed Solution

- Rename `read_redcap_tidy()` to `import_redcap()`. Have `read_redcap_tidy()` throw a deprecation warning (will be removed in 0.3) and call `import_redcap()`
- Rename `bind_tables()` to `bind_tibbles()`. Deprecate the old function as above.
- Rename `extract_table()` to `extract_tibble()`. Deprecate the old function as above.
- Rename `extract_tables()` to `extract_tibbles()`. Deprecate the old function as above.
- For `bind_tibbles()`, rename `.data` to `supertbl`. Remove `structure` (not a useful option IMO)
- For `extract_tibble()` and `extract_tibbles()`, rename `.data` to `supertbl`
- Update tests to reflect these changes
- Update roxygen documentation for exported functions
- Update README and vignettes with new function names, including screenshots/GIFs

### Checklist

- [x] The issue is atomic
- [x] The issue description is documented
- [x] The issue title describes the problem succinctly
- [x] Developers are assigned to the issue
- [x] Labels are assigned to the issue

### PR DESCRIPTION
**Pull request by skadauke**
# Description

This PR refactors the package to rename `read_redcap_tidy()` to `import_redcap()`, `bind_tables()` to `bind_tibbles()`, `extract_table()` to `extract_tibble()`, and `extract_tables()` to `extract_tibbles()`. 

The approach I took to do this refactor safely is the following:

1. Rename the function's `.R` and test file.
2. Edit test file to use the new function name (and if appropriate remove obsolete tests). This should break a bunch of tests.
3. Edit the `.R` file until all tests work again. Implement any changes to the API.
4. Create a deprecated function in `R/deprecated.R` that either calls the new function or simply re-implements the old one. Add a deprecation warning to that function. Follow this example: https://github.com/tidyverse/tibble/blob/HEAD/R/deprecated.R.
5. Find all instances of the function's name inside the package (docs, etc.) and replace if appropriate.
6. Write a basic test to make sure the deprecated and the new function produce the same output.

# Proposed Changes 

- [x] Rename read_redcap_tidy() to import_redcap(). Rename the `suppress_messages` argument to `suppress_redcapr_messages` for clarity about which messages we'll suppress
- [x] Rename bind_tables() to bind_tibbles(). Remove the `structure` argument. Rename the `.data` argument to `supertbl`.
- [x] Rename extract_table() to extract_tibble(). Rename the `.data` argument to `supertbl`.
- [x] Rename extract_tables() to extract_tibbles(). Rename the `.data` argument to `supertbl`.
- [x] Review the documentation and recreate screenshots as necessary.

### Issue Addressed
**Closes #91**

### PR Checklist

Before submitting this PR, please check and verify below that the submission meets the below criteria:

- [x] New/revised functions have associated tests
- [x] New/revised functions that update downstream outputs have associated static testing files (`.RDS`) updated under `inst/testdata/create_test_data.R`
- [x] New tests that make API calls use `httptest::with_mock_api` and any new mocks were added to `tests/testthat/fixtures/create_httptest_mocks.R`
- [x] New/revised functions use appropriate naming conventions
- [x] New/revised functions don't repeat code
- [x] Code changes are less than **250** lines total
- [x] Issues linked to the PR using [GitHub's list of keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [x] The appropriate reviewer is assigned to the PR
- [x] The appropriate developers are assigned to the PR

# Code Review
This section to be used by the reviewer and developers during Code Review after PR submission

### Code Review Checklist

- [ ] I checked that new files follow naming conventions and are in the right place
- [ ] I checked that documentation is complete, clear, and without typos
- [ ] I added/edited comments to explain "why" not "how"
- [ ] I checked that all new variable and function names follow naming conventions
- [ ] I checked that new tests have been written for key business logic and/or bugs that this PR fixes
- [ ] I checked that new tests address important edge cases

### CODE REVIEW

File: tests/testthat/test-deprecated.R [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+test_that("read_redcap_tidy and import_redcap produce the same output", {
+
+  httptest::with_mock_api({
+    out_1 <-
+      import_redcap(redcap_uri, longitudinal_token)
+
+    out_2 <-
+      read_redcap_tidy(redcap_uri, longitudinal_token) %>%
+      suppressWarnings() # Suppress deprecation warning
```
Comment (ezraporter) [inline tests/testthat/test-deprecated.R:14]: Can specify the class here
```suggestion
      suppressWarnings(classes = "lifecycle_warning_deprecated") # Suppress deprecation warning
```

File: tests/testthat/test-deprecated.R [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  redcaptidier_longitudintal_db %>%
+    bind_tibbles(tbls = c("nonrepeated", "repeated"),
+                 environment = env_1)
+
+  env_2 <- new_environment()
+
+  redcaptidier_longitudintal_db %>%
+    bind_tables(tbls = c("nonrepeated", "repeated"),
+                 environment = env_2) %>%
+    suppressWarnings() # Suppress deprecation warning
```
Comment (ezraporter) [inline tests/testthat/test-deprecated.R:39]:
```suggestion
    suppressWarnings(classes = "lifecycle_warning_deprecated") # Suppress deprecation warning
```

File: tests/testthat/test-deprecated.R [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+      "testdata/redcaptidier_longitudinal_db.RDS", package = "REDCapTidieR"
+    )
+  )
+
+  out_1 <- redcaptidier_longitudintal_db %>%
+    extract_tibble(tbl = "nonrepeated")
+
+  out_2 <- redcaptidier_longitudintal_db %>%
+    extract_table(tbl = "nonrepeated") %>%
+    suppressWarnings() # Suppress deprecation warning
```
Comment (ezraporter) [inline tests/testthat/test-deprecated.R:58]:
```suggestion
    suppressWarnings(classes = "lifecycle_warning_deprecated") # Suppress deprecation warning
```

File: tests/testthat/test-deprecated.R [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+      "testdata/redcaptidier_longitudinal_db.RDS", package = "REDCapTidieR"
+    )
+  )
+
+  out_1 <- redcaptidier_longitudintal_db %>%
+    extract_tibbles(tbls = starts_with("non"))
+
+  out_2 <- redcaptidier_longitudintal_db %>%
+    extract_tables(tbls = starts_with("non")) %>%
+    suppressWarnings() # Suppress deprecation warning
```
Comment (ezraporter) [inline tests/testthat/test-deprecated.R:77]:
```suggestion
    suppressWarnings(classes = "lifecycle_warning_deprecated") # Suppress deprecation warning
```

File: R/deprecated.R [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+  out <- .data %>%
+    select("redcap_form_name", "redcap_data") %>%
+    pivot_wider(names_from = "redcap_form_name",
+                values_from = "redcap_data")
+
+  out <- out[eval_select(tbls, data = out)]
+
+  out %>%
+    map(.f = ~pluck(.)[[1]])
```
Comment (ezraporter) [inline R/deprecated.R:112]:
```suggestion
  extract_tibbles(supertbl = .data, tbls = tbls)
```

**Review by ezraporter — CHANGES_REQUESTED** [review]
Looks pretty good! I don't think we need to reimplement `extract_tables()` within `extract_tibbles()` in `deprecated.R` unless I'm missing something. Requesting changes for that.

Other minor suggestions in the comments as well.

File: R/import_redcap.R [hunk 6]
```diff
@@ -66,12 +63,12 @@
...
 #' @export
 
-read_redcap_tidy <- function(redcap_uri,
+import_redcap <- function(redcap_uri,
                              token,
                              raw_or_label = "label",
                              forms = NULL,
                              export_survey_fields = TRUE,
-                             suppress_messages = TRUE) {
+                             suppress_redcapr_messages = TRUE) {
```
[the remark this answers is not in the document]
Comment (skadauke) [reply R/import_redcap.R:71]: I think this is OK as is - it's specifically because REDCapR has an annoying idiosyncratic default behavior of throwing a message with each API call.

File: vignettes/REDCapTidieR.Rmd [hunk 7]
```diff
@@ -46,29 +46,29 @@ Here is a screenshot of the REDCap Record Status Dashboard of this database. It
...
 
-Each row of the supertibble corresponds to one REDCap instrument. The `redcap_form_name` and `redcap_form_label` columns identify that instrument. The `redcap_data` column contains the data tibbles. The `redcap_metadata` column contains the metadata tibbles. Additional columns contain useful information about the data tibble, such as the number of rows and columns, the size in memory, and the percentage of missing values.
+**Each row of the supertibble corresponds to one REDCap instrument**. The `redcap_form_name` and `redcap_form_label` columns identify which instrument the row relates to. The `redcap_data` column contains the data tibbles. The `redcap_metadata` column contains the metadata tibbles. Additional columns contain useful information about the data tibble, such as the number of its rows and columns, its size in memory, and the percentage of missing values in the data.
```
[the remark this answers is not in the document]
Comment (skadauke) [reply vignettes/REDCapTidieR.Rmd:67]: I'm not sure there is really an inconsistency. I think REDCap uses "form" and "instrument" interchangeably. The REDCap API uses the term "form" so for consistency it's good we use "form" in arguments and in variable names. But I think the term "instrument" has a more specific meaning than "form" so I prefer to use "instrument" in the documentation narrative.

File: tests/testthat/test-deprecated.R (hunk 1 above)
Comment (skadauke) [reply tests/testthat/test-deprecated.R:14]: Done! Also the below.

File: R/deprecated.R (hunk 5 above)
Comment (skadauke) [reply R/deprecated.R:112]: Huh, I thought I tried exactly this but ran into errors because of forwarding tidyselect functions. But your code works!

File: README.Rmd [hunk 8]
```diff
@@ -49,9 +49,9 @@ devtools::install_github("CHOP-CGTInformatics/REDCapTidieR")
 
 ## Getting Started
 
-Use `read_redcap_tidy()` together with `bind_tables()` to import data from all instruments into your environment.
+Use `import_redcap()` together with `bind_tibbles()` to import data from all instruments into your environment.
 
-![](man/figures/redcaptidier_demo.gif)
+![](man/figures/redcaptidier-demo.gif)
```
[the remark this answers is not in the document]
Comment (skadauke) [reply README.Rmd:54]: Great point! Added to all docs in the next commit.

**Review by ezraporter — APPROVED** [review]
